sdk/personalizer/azure-ai-personalizer/src/main/java/com/azure/ai/personalizer/PersonalizerAdminAsyncClient.java

```diff
@@ -0,0 +1,376 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.personalizer;
+
+import com.azure.ai.personalizer.implementation.PersonalizerClientV1Preview3Impl;
+import com.azure.ai.personalizer.implementation.PersonalizerClientV1Preview3ImplBuilder;
+import com.azure.ai.personalizer.implementation.models.Evaluation;
+import com.azure.ai.personalizer.implementation.models.EvaluationContract;
+import com.azure.ai.personalizer.implementation.models.LogsProperties;
+import com.azure.ai.personalizer.implementation.models.ModelProperties;
+import com.azure.ai.personalizer.implementation.models.PolicyContract;
+import com.azure.ai.personalizer.implementation.models.PolicyReferenceContract;
+import com.azure.ai.personalizer.implementation.models.ServiceConfiguration;
+import com.azure.ai.personalizer.implementation.util.Transforms;
+import com.azure.core.annotation.ReturnType;
+import com.azure.core.annotation.ServiceMethod;
+import com.azure.core.http.rest.Response;
+import com.azure.core.http.rest.SimpleResponse;
+import com.azure.core.http.rest.StreamResponse;
+import com.azure.core.util.Context;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.FluxUtil;
+import com.azure.core.util.logging.ClientLogger;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.nio.ByteBuffer;
+import java.util.List;
+
+import static com.azure.core.util.FluxUtil.monoError;
+import static com.azure.core.util.FluxUtil.withContext;
+
+public final class PersonalizerAdminAsyncClient {
+
+    private PersonalizerClientV1Preview3Impl service;
+    private final ClientLogger logger = new ClientLogger(PersonalizerAdminAsyncClient.class);
+
+    public PersonalizerAdminAsyncClient(PersonalizerClientV1Preview3Impl service) {
+    PersonalizerClientV1Preview3ImplBuilder builder = new PersonalizerClientV1Preview3ImplBuilder();
+        this.service = service;
+    }
+
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Evaluation> createEvaluation(EvaluationContract evaluationContract) {
+        return createEvaluationWithResponse(evaluationContract).flatMap(FluxUtil::toMono);
+    }
+
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<Evaluation>> createEvaluationWithResponse(EvaluationContract evaluationContract) {
+        try {
+            return withContext(context -> createEvaluationWithResponse(evaluationContract, context));
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    Mono<Response<Evaluation>> createEvaluationWithResponse(EvaluationContract evaluationContract, Context context) {
+        if (evaluationContract == null) {
+            throw logger.logExceptionAsError(new IllegalArgumentException("'evaluationContract' is required and cannot be null"));
```

Comment:
```suggestion
Objects.requireNonNull(evaluationContract, "'evaluationContract' is required and cannot be null")
```

Reply:
Do we need the entire model of evaluationContract or just one of the properties are required for a successful API?

Reply:
The entire contract is required.